File: main/src/main/java/com/sedmelluq/discord/lavaplayer/player/event/TrackStartEvent.java
package com.sedmelluq.discord.lavaplayer.player.event;

import com.sedmelluq.discord.lavaplayer.player.AudioPlayer;
import com.sedmelluq.discord.lavaplayer.track.AudioTrack;

/**
 * Event that is fired when a track starts playing.
 */
public class TrackStartEvent extends AudioEvent {
  /**
   * Audio track that started
   */
  public final AudioTrack track;

  /**
   * @param player Audio player
   * @param track Audio track that started
   */
  public TrackStartEvent(AudioPlayer player, AudioTrack track) {
    super(player);
    this.track = track;
  }
}

File: main/src/main/java/com/sedmelluq/discord/lavaplayer/manager/event/TrackStartEvent.kt
package com.sedmelluq.discord.lavaplayer.manager.event

import com.sedmelluq.discord.lavaplayer.manager.AudioPlayer
import com.sedmelluq.discord.lavaplayer.track.AudioTrack

/**
 * Event that is fired when a track starts playing.
 *
 * @param player Audio player
 * @param track  Audio track that started
 */
data class TrackStartEvent(
    override val player: AudioPlayer,
    /**
     * Audio track that started
     */
    val track: AudioTrack
) : AudioEvent
